tests/cypress/scripts/cliHelper.js

```diff
@@ -68,94 +255,19 @@ export const cliHelper = {
         return cy.wrap(targetCluster)
       })
   },
-  generateNamespace: (prefix, postfix) => {
-    return `${prefix ? prefix : 'search'}-${postfix ? postfix : Date.now()}`
-  },
-  createNamespace: (name, kubeconfig = '') => {
-    cy.exec(`${kubeconfig} oc get namespace ${name}`, {
-      failOnNonZeroExit: false,
-    }).then((res) => {
-      if (!res.stderr) {
-        cy.log(`Namespace: ${name} already exist within the cluster`)
-      } else {
-        cy.log(
-          `Namespace: ${name} does not exist within the cluster. Preparing to create namespace resource.`
-        )
-        cy.exec(`${kubeconfig} oc create namespace ${name}`)
-      }
-    })
-  },
-  createDeployment: (name, namespace, image, kubeconfig = '') => {
-    cy.exec(
-      `${kubeconfig} oc create deployment ${name} --image=${image} -n ${namespace}`
-    ).then((res) => {
-      cy.log(res.stdout ? res.stdout : res.stderr)
-    })
-  },
-  createApplication: (appName, namespace) => {
-    cy.readFile('tests/cypress/templates/application.yaml').then((cfg) => {
-      let b64Cfg = btoa(
-        cfg.replaceAll('APPNAME', appName).replaceAll('NAMESPACE', namespace)
-      )
-      cy.exec(`echo ${b64Cfg} | base64 -d | oc apply -f -`)
-      cy.log(`Successfully created application (${appName})`)
-    })
-  },
-  deleteNamespace: (name, kubeconfig = '') => {
-    cy.exec(`${kubeconfig} oc delete namespace ${name}`).then((res) => {
-      cy.log(res.stdout ? res.stdout : res.stderr)
-    })
-  },
-  login: (mode) => {
-    var mode = mode === 'Local' ? 'HUB' : 'MANAGED'
+
+  /**
+   * Login into the cluster environment with the `oc` cli command.
+   * @param {string} cluster The cluster environment to login into (Default: HUB).
+   * @param {object} options Additional options for logging into the cluster environment.
+   */
+  login: (cluster = 'HUB', options = {}) => {
     cy.exec(
       `oc login --server=https://api.${Cypress.env(
-        `OPTIONS_${mode}_BASEDOMAIN`
-      )}:6443 -u ${Cypress.env(`OPTIONS_${mode}_USER`)} -p ${Cypress.env(
-        `OPTIONS_${mode}_PASSWORD`
+        `OPTIONS_${cluster}_BASEDOMAIN`
+      )}:6443 -u ${Cypress.env(`OPTIONS_${cluster}_USER`)} -p ${Cypress.env(
+        `OPTIONS_${cluster}_PASSWORD`
       )} --insecure-skip-tls-verify`
```

Comment:
A login error here won't make the test to fail. That will be problematic to debug later because execution will fail at a later step and it might not be clear that it's because the login failed. It's better to validate here and throw an error with a clear explanation of why the test failed.

Reply:
Also, because this is called from `before()` it won't retry if there's an error. I wonder if it's worth adding some retry logic here.